### ISSUE TITLE
Rename shortcut should be disabled for non-HNS blob containers

### ISSUE THREAD

**Issue by tjohnson1010** [microsoft/AzureStorageExplorer#8056]
### Preflight Checklist

- [X] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.34.0 (99)

### Architecture

x64

### Storage Explorer Build Number

20240532.2

### Platform

Windows

### OS Version

Windows 11

### Bug Description

When renaming a blob, it creates a new file with the new name but doesn't remove the old file.

### Steps to Reproduce

1. Launch Storage Explorer
2. Go to any Blob Container
3. Click an file.
4. Press F2
5. Enter new name and click Enter

### Actual Experience

Two files exist, one with old name and one with new name.

### Expected Experience

Only one file should exist with the new name.

### Additional Context

This feature works with files in File Shares but not Blob Containers

**Comment by craxal** [comment]
@tjohnson1010 You're using ADLS Gen2 Blob Storage, right? This seems odd, as we are using a move function provided by the Azure SDK. Do you get any error messages at all? Does this happen consistently?

**Comment by tjohnson1010** [comment]
No error and happens all the time.

Below are settings for the two I teste with.  One was  gen1 the other gen2.  I rarely use rename with Blob containers, so no bug deal.  I use rename mostly with file shares and that works 100%.


FIRST Storage: (doesn’t work)
***@***.***
***@***.***


SECOND Storage: (doesn’t work)
***@***.***
***@***.***


1 of many File Share I use this on:  (works)
***@***.***
***@***.***

Have a good day,

Thomas Johnson
Director of IT Applications

Phone: 630.242.3385
***@***.******@***.***>  |  www.legacyscs.com<https://legacyscs.com/>

distribution | fulfillment | transportation

This confidential e-mail and any attachments are for the exclusive use of the addressee(s). If you receive this in error, please delete all copies and notify the sender. Any views expressed are those of the sender, unless stated otherwise.

**Comment by craxal** [comment]
@tjohnson1010 Were you trying to post screenshots? Can you do it again using the GitHub website? You can't do it replying via email.

**Comment by tjohnson1010** [comment]
No  I pasted info about the types of Blobs I've tested with.  One was gen2 the other was gen1.  Both had same behavior.

**Comment by craxal** [comment]
The move operation does not appear to have a corresponding REST API, so it could be a bug in the Azure client SDK.

Can you do the following:
1. Launch Storage Explorer.
2. Reproduce the issue.
3. Share the app logs?

You can also try using a network monitoring app to capture the actual REST APIs that Storage Explorer uses and see if there are any error responses that come back.

**Comment by tjohnson1010** [comment]
OK.  I see what is happening now.  I notice under Blobs there is no Rename option.  It only exists for File Shares.  But The F2 key exists for both types.  For Filke Shares it does rename but for Blobs it's doing a Copy (the pop-up states Clone).  There is no rename feature for Blob.  To be consistent with other MS apps, like Explorer, F2 shouldn't do anything for Blob since it doesn't support Rename.

**Comment by craxal** [comment]
@tjohnson That's not quite true. ADLS Gen2 blobs allow for hierarchical namespaces (real folders, just like file shares). So, Rename/Move is a perfectly valid operation.

Here's an ADLS Gen2 blob container's toolbar:
<img width="1168" alt="image" src="https://github.com/user-attachments/assets/e261321a-7f8a-4977-b91f-58a7df8f16b9">

And here's a non-ADLS Gen2 blob container's toolbar:
<img width="1168" alt="image" src="https://github.com/user-attachments/assets/660adf05-2446-4797-aaa5-5e6c05e0475f">

I should note that ADLS Gen1 is retired as of February 2024. So, maybe you're referring to something else when you mentioned Gen1 earlier?

**Comment by tjohnson1010** [comment]
I see now I had Account kind of General Purpose V2 Type and Hierarchical namespace disabled.  So ADLS wasn't enabled.
I'm ok with no Rename on non ADLS, just the F2 having different functionality for the different toolbars threw me for a loop.

**Comment by craxal** [comment]
@tjohnson1010 You make a very good point. Keyboard shortcuts should be consistent. We'll review as a team and address as necessary.

**Comment by craxal** [comment]
Just for clarity, F2 will _rename_ the blob if it is ADLS Gen2 and _clone_ if not. That is explicitly in the code. It's unclear how disruptive removing the F2 shortcut for cloning will be to other users.

**Comment by craxal** [comment]
After some team discussion, we decided not to alter the existing shortcuts. The shortcuts have been in place for quite some time, and we don't want to disrupt other users' workflows.

Historically, clone started out as a client-side rename. But we had concerns about errors leading to data loss, and we adopted the policy of not implementing destructive operations unless the platform specifically had an API for doing that operation. So instead of renaming, we went to cloning for non-ADLS Gen2 blobs.
